demos/http3/ossl-nghttp3-demo-server.c

```diff
@@ -130,15 +178,127 @@ static int on_end_stream(nghttp3_conn *h3conn, int64_t stream_id,
     return 0;
 }
 
-static int read_from_ssl_ids(nghttp3_conn *h3conn, struct h3ssl *h3ssl)
+/* Read from the stream and push to the h3conn */
+static int quic_server_read(nghttp3_conn *h3conn, SSL *stream, uint64_t id, struct h3ssl *h3ssl)
 {
+    int ret, r;
     uint8_t msg2[16000];
+    size_t l = sizeof(msg2);
+
+    if (!SSL_net_read_desired(stream))
+        return 0; /* Nothing to read */
+
+    ret = SSL_read(stream, msg2, l);
```

Comment:
This does not look right. IMO instead of SSL_net_read_desired(), what you want to use is SSL_has_pending() here.

Reply:
fixed.